### ISSUE TITLE
Null reference exception after forgetting to call configuration extension method

### ISSUE THREAD

**Issue by NickLydon** [testcontainers/testcontainers-dotnet#428]
**Is your feature request related to a problem? Please describe.**
I missed the call to `WithKafka(...)` when creating my container:
```csharp
await using var kafkaTestcontainer = new TestcontainersBuilder<KafkaTestcontainer>()
  // .WithKafka(new KafkaTestcontainerConfiguration())
  .Build();
await kafkaTestcontainer.StartAsync();
```
This resulted in a null reference exception, which wasn't clear:

> System.NullReferenceException: Object reference not set to an instance of an object.
> 
> System.NullReferenceException
> Object reference not set to an instance of an object.
>    at DotNet.Testcontainers.Clients.TestcontainersClient.RunAsync(ITestcontainersConfiguration configuration, CancellationToken ct)
>    at DotNet.Testcontainers.Containers.TestcontainersContainer.Create(CancellationToken ct)
>    at DotNet.Testcontainers.Containers.TestcontainersContainer.StartAsync(CancellationToken ct)
>    at Xunit.Sdk.ExceptionAggregator.RunAsync(Func`1 code) in C:\Dev\xunit\xunit\src\xunit.core\Sdk\ExceptionAggregator.cs:line 90

**Describe the solution you'd like**
The simplest solution would be to do some null-checking and throw more detailed exceptions, or to throw if the configuration is completely unchanged by the user.

**Describe alternatives you've considered**
Alternatively it would be nice if the API didn't need both the type signature (`TestcontainersBuilder<KafkaTestcontainer>`) as well as a separate extension method. A single place to configure it would be simpler. 

**Additional context**
It's a minor issue really, but it came up after a long time wrestling with a new M1 Macbook trying to get the kafka client library working. I assumed at first it was due to that and not to incorrect usage of the testcontainers library. 
P.S. thanks for this great library!

**Comment by HofmeisterAn** [comment]
https://github.com/HofmeisterAn/dotnet-testcontainers/blob/b99957969f1bf8d1615196f1a77899cb5d51c76e/src/DotNet.Testcontainers/Builders/TestcontainersBuilder.cs#L298

Does not check for mandatory properties (a check for `image` should be enough).

**Comment by HofmeisterAn** [comment]
Released in `1.6.0-beta.2126` (#429).
